R/S3.R

```diff
@@ -241,3 +313,36 @@ class_data.frame <- new_S3_class("data.frame",
   },
   validator = validate_data.frame
 )
+
+#' @export
+#' @rdname base_s3_classes
+#' @format NULL
+#' @order 3
+class_matrix <- new_S3_class("matrix",
+  constructor = function(.data = NA, nrow = 1, ncol = 1, byrow = FALSE, dimnames  = NULL) {
```

Comment:
Maybe `nrow` can default to `length(.data)`?  

The `nrow = 1` default in `base::matrix()` is a bit of a lie, because there is also a `missing()` check in the C code. The current constructor emits a warning and drops vector elements, given the convenient case of wanting to turn an length>1 atomic vector into a 1-col matrix.  

E.g,
``` r
matrix(1:4)
#>      [,1]
#> [1,]    1
#> [2,]    2
#> [3,]    3
#> [4,]    4
matrix(1:4, 1, 1)
#> Warning in matrix(1:4, 1, 1): data length differs from size of matrix: [4 != 1
#> x 1]
#>      [,1]
#> [1,]    1


constructor = function(.data = NA, nrow = 1, ncol = 1, byrow = FALSE, dimnames = NULL) {
  matrix(.data, nrow, ncol, byrow, dimnames)
}

constructor(1:4)
#> Warning in matrix(.data, nrow, ncol, byrow, dimnames): data length differs from
#> size of matrix: [4 != 1 x 1]
#>      [,1]
#> [1,]    1
```